src/hooks/useValidateRepository.ts

```diff
@@ -0,0 +1,188 @@
+import {useCallback} from 'react';
+import {useDebounce} from 'react-use';
+
+import {NamePath} from 'antd/lib/form/interface';
+
+import {dummySecret} from '@utils/sources';
+
+const fieldsNames: string[] = ['uri', 'token', 'username', 'branch', 'path'];
+
+const authSearchString = 'username or token:';
+const uriSearchString = 'uri:';
+const branchSearchString = 'branch:';
+const pathSearchString = 'path:';
+const endSearchString = ', context';
+
+const getErrorMessage = (rawErrorString: string, searchString: string) =>
+  rawErrorString.slice(
+    rawErrorString.search(searchString) + searchString.length,
+    rawErrorString.search(endSearchString)
```

Comment:
`String.prototype.search` is using `RegExp`. There you want to search for the actual string, so `indexOf` is more appropriate:
```suggestion
    rawErrorString.indexOf(searchString) + searchString.length,
    rawErrorString.indexOf(endSearchString)
```

Overall, parsing the message string feels hacky a lot, so most likely it will have some bugs (i.e. because of`endSearchString` found before `searchString`)

Reply:
Accepted indexOf change, will talk about this parsing with Vlad